students/ThanhNhan/lesson06/mailroom4.py

```diff
@@ -0,0 +1,251 @@
+    """
+
+    if (name == " " or name.isdigit()):
+        return False
+
+    str_name = name.rstrip()
+    # #parse first and last name into string
+    try:
+        str_name = str_name.split(' ') 
+        if (str_name[1] == ' '):    
+            return False
+    except IndexError:
+        print ("Need to have first and last name within space between")
+        return False
+
+    #Validate user not to enter digit  
+    print(str_name)
+    for char in str_name:  
+        for i in char:
+            if (i.isnumeric() == True):
+                print("Must be Character")
+                return False
+    return True
+
+
+def lookup_name(name):
+    """
+    This method is looking up user name
+
+    params:  take name of the user
+    returns when user is found from the list or None
+    """
+    #Set to lower case for comparison
+
+    for x in donor_dict.keys():
+        if (x.split(' ')[0].lower() == name.split(' ')[0].lower() and 
+            x.split(' ')[1].lower() == name.split(' ')[1].lower()):
+            return x
+    
+    return None
+
+
+def create_letter(name, amount):
+    """
+    This method is generating template letter to who addressing to 
+    and the amount of donation
+
+    params:  take user donor dictionary data
+    returns whole string of the template letter
+    """
+    strLetter = """
+            Dear {},
+            Thank you  for your very kind donation of ${:.2f}
+            It will be put to very good use.
+
+                            Sincerely,
+                            - The Team   
+	        """.format(name, amount)
+    return strLetter
+
+def str_lname(rpt_donor):
+    """
+    This method is to return the last name from the string
+
+    params:  string value of person last name
+    returns whole string of the template letter
+    """
+    print (rpt_donor[0].split(' ')[1])
+    return rpt_donor[0].split(' ')[1]
+
+
+def create_report():
+
+	# Create new list that contain name, amount, donate how many times, average
+    
+    report_donor = []
+
+    for key, values in donor_dict.items():    
+        name = "{}".format(key)
+        total_cost = sum(values)
+        n_time = len(values)
+        average = total_cost/n_time
+        report_donor.append([name, round(total_cost, 2), n_time, round(average, 2)])
+
+    print (report_donor)
+    #sort the report
+    report = sorted(report_donor, key=str_lname, reverse=False)
+
+    print("{:20}|  {:<10s}|  {:<10s}|  {:<10s} ".format(*title))
+    print("".join(["-"] * 62))
+    for x in range (len(report)):
+    	print("{:20} $ {:10.2f} {:>12d} $ {:>12.2f}".format(*report[x]))
+    
+    return report_donor
+ 
+def quit_program():
+    print("Good Bye!")
+    sys.exit()  # exit the interactive script
+
+def ck_dir():
+    if not os.path.isdir(DIR_NAME):
+        os.mkdir(DIR_NAME)
+        return False
+    return True
+
+def ss_letterall():
+#     """
+#     This method write letter to disk as txt file
+
+#     params:  none
```

Comment:
no need to specify None for the params -- you can tell by looking at the function definition.